module/data/activity/base-activity.mjs

```diff
@@ -508,6 +511,55 @@ export default class BaseActivityData extends foundry.abstract.DataModel {
   /*  Helpers                                     */
   /* -------------------------------------------- */
 
+  /**
+   * Get the roll parts used to create the damage rolls.
+   * @param {Partial<DamageRollProcessConfiguration>} [config={}]
+   * @returns {DamageRollProcessConfiguration}
+   */
+  getDamageConfig(config={}) {
+    if ( !this.damage?.parts ) return foundry.utils.mergeObject({ rolls: [] }, config);
+
+    const scaling = config.scaling ?? 0; // TODO: Set properly once scaling is handled during activation
+    const rollConfig = foundry.utils.mergeObject({ scaling }, config);
+    rollConfig.rolls = this.damage.parts
+      .map(d => this.processDamagePart(d, rollConfig))
+      .filter(d => d.parts.length)
+      .concat(config.rolls ?? []);
+
+    return rollConfig;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Process a single damage part into a roll configuration.
+   * @param {DamageData} damage                          Damage to prepare for the roll.
+   * @param {DamageRollProcessConfiguration} rollConfig  Roll configuration being built.
+   * @returns {DamageRollConfiguration}
+   */
+  processDamagePart(damage, rollConfig) {
+    const scaledFormula = damage.scaledFormula(rollConfig.scaling);
+    const parts = scaledFormula ? [scaledFormula] : [];
+    const data = this.getRollData();
```

Comment:
Not great that we have to call this on every part, but I don't know if it can be helped if this is intended to be a public method. Otherwise if it's only called as part of `getDamageConfig`, we can pass it in from there.